models/stock_move_line.py

```diff
@@ -1,7 +1,55 @@
-from odoo import models, fields
+# -*- coding: utf-8 -*-
+
+from odoo import api, models, fields, _
+from odoo.exceptions import ValidationError
 
 
 class StockMoveLine(models.Model):
     _inherit = 'stock.move.line'
 
     x_selected = fields.Boolean(string=' ', help='Check this box to select')
+
+    u_result_parent_package_id = fields.Many2one('stock.quant.package',
+                    'Parent Destination Package', ondelete='restrict')
+
+    def _action_done(self):
+        """ When a move_line is done and it has result_package_id its
+            parent will be removed if u_result_parent_package_id is empty,
+            otherwise it will be updated to be u_result_parent_package_id.
+        """
+        super(StockMoveLine, self)._action_done()
+
+        for ml in self:
+            result_parent = ml.u_result_parent_package_id
+            result_package = ml.result_package_id
+            if result_parent:
+                if result_package:
+                    # only update if it is different
+                    if result_package.package_id != result_parent:
+                        result_package.package_id = result_parent
+                else:
+                    raise ValidatinError(
+                            _('Result parent package without result'
+                              ' package at picking %s') % ml.picking_id.name)
+            else:
+                if result_package and result_package.package_id:
+                    result_package.package_id = False
+
+    @api.constrains('u_result_parent_package_id')
+    @api.onchange('u_result_parent_package_id')
+    def _assert_one_parent_package(self):
```

Comment:
Did you consider using a groupby approach here to find any result_package with more than one result_parent? something like this?
`
        StockMoveLine = self.env['stock.move.line']
        by_res_pack = lambda ml: ml.result_package_id.id
        res_parents_by_res_pack = {res_pack: StockMoveLine.union(*move_lines).mapped('u_result_parent_package_id')
                                   for res_pack, move_lines in
                                   groupby(self.sorted(key=by_res_pack), key=by_res_pack)}
        errors = []
        for pack, parents in res_parents_by_res_pack.items():
            if len(parents) > 1:
                errors.append("%s -> %s" % (pack.name, ', '.join(parents.mapped('name'))))
        if errors:
            raise ValidationError(
                    _('Multiple result parents found for '
                      'packages:\n%s') % '\n'.join(errors))`

This allows us to show all the errors at once rather than just the first one found, and it's also probably faster for large pickings.

Reply:
Nope, as I was assuming that we will only have only one failing move_line at this point, because onchange/depends are not allowing it.

Reply:
Ah, fair point. Fine as it is then.